framework/include/meshgenerators/SideSetsGeneratorBase.h

```diff
@@ -52,13 +52,42 @@ class SideSetsGeneratorBase : public MeshGenerator
    */
   void flood(const Elem * elem, Point normal, boundary_id_type side_id, MeshBase & mesh);
 
-  boundary_id_type getNextBoundaryID() const;
+  bool normalsWithinTol(const Point & normal_1, const Point & normal_2) const;
 
-  Real _variance;
-  bool _fixed_normal;
+  bool addSideToBoundary(const Elem * elem, const unsigned int side) const;
+
+  /// the mesh to add the sidesets to
+  std::unique_ptr<MeshBase> & _input;
+
+  const bool _fixed_normal;
 
   /// Whether or not to remove the old sidesets (if any) when adding sidesets
-  bool _replace;
+  const bool _replace;
+
+  /// Whether to only include external side when considering sides to add to the sideset
+  const bool _include_only_external_sides;
+
+  /// Whether to skip side if it is already part of existing sideset
+  const bool _skip_if_part_of_existing_sideset;
+
+  /// Whether to only include if side is already part of existing sideset
+  const bool _include_only_if_part_of_existing_sideset;
+
+  ///// Whether to only include if side is touching a subdomain
+  // const bool _include_only_if_touching_subdomain;
+
+  /// if specified, then faces are only added if their normal is close to this
+  Point _normal;
+
+  /// true if only faces close to "normal" will be added
+  bool _using_normal;
```

Comment:
yup this is all great consolidation

Reply:
Should we be concerned that not every one of these variables applies to every derived class? For instance, `_include_only_external_sides` is used in `ParsedGenerateSideset` and `SideSetsAroundSubdomainGenerator`, but not in `AllSideSetsByNormalsGenerator`, `SideSetsFromNormalsGenerator`, or `SideSetsFromPointsGenerator`. As currently implemented, the later three always only include external sides, so there is no need to check the value of `_include_only_external_sides` in these generators. In some cases, it could make sense to incorporate `_include_only_external_sides` into these generators. For instance, in `AllSideSetsByNormalsGenerator`, if `_include_only_external_sides` is set to false and a subdomain id is specified, the result would look very similar to something from `SideSetsAroundSubdomainGenerator`. Do we want to make these kind of modifications, or just be okay with some derived classes not using some of the base class parameters?

Reply:
it s ok to have derived class remove parameters.
set them to the decent value then use suprressParam

Reply:
We should also consider if we have any other sideset-generating class that can inherit from this base class and gain capability. Maybe not

Reply:
> Do we want to make these kind of modifications

if we can gain meaningful capability at little efforts I would say yes. But usually it s best to be needs-driven. As you are generating the mesh you realize you want this, and because it s in the base class it's very clean to add it

Reply:
> We should also consider if we have any other sideset-generating class that can inherit from this base class and gain capability. Maybe not

Yeah, the base class could also be applicable to the following generators. I'll check before I'm ready to close.

SideSetsFromNodeSetsGenerator
SideSetsFromBoundingBoxGenerator
RefineSidesetGenerator
SideSetsBetweenSubdomainsGenerator